drivers/storage/portworx/portworx.go

```diff
@@ -219,6 +219,13 @@ func (p *portworx) SetDefaultsOnStorageCluster(toUpdate *corev1.StorageCluster)
 			toUpdate.Status.DesiredImages.MetricsCollectorProxy = release.Components.MetricsCollectorProxy
 		}
 
+		if autoUpdatePxRepo(toUpdate) &&
```

Comment:
Can we add unit tests for this? You can refer the tests here - https://github.com/libopenstorage/operator/blob/master/drivers/storage/portworx/portworx_test.go#L575.